src/java.desktop/windows/classes/sun/awt/windows/ThemeReader.java

```diff
@@ -111,11 +131,15 @@ private static Long getTheme(String widget) {
             try {
                 if (!valid) {
                     // Close old themes.
-                    for (Long value : widgetToTheme.values()) {
-                        closeTheme(value);
+                    if (!dpiAwareWidgetToTheme.isEmpty()) {
+                        for (Long value :
+                             dpiAwareWidgetToTheme.get(dpi).values()) {
```

Comment:
I think it's still possible that `get(dpi)` returns `null`, you should handle `null` gracefully.